src/app/pages/questions/components/finish/finish.component.ts

```diff
@@ -43,22 +44,24 @@ export class FinishComponent implements OnChanges {
   displayNextTaskReminder = true
   completedInClinic = false
   shadowStyle = 'inset 100px 0 0 -50px #0B4A59'
+  progressDisplay = 0
+  startTime = Date.now()
+  etaText = ''
 
   constructor(
     private usage: UsageService,
     private localization: LocalizationService
   ) {}
 
   ngOnChanges() {
+    if (this.isShown) this.usage.setPage(this.constructor.name)
+    if (this.progressCount == 1) this.showDoneButton = true
     this.displayNextTaskReminder =
       this.taskType == AssessmentType.SCHEDULED && !this.isLastTask
-    this.innerText = this.localization.translateKey(LocKeys.SETTINGS_WAIT_ALERT) + '...'
-    if (this.isShown) this.usage.setPage(this.constructor.name)
-    this.shadowStyle = `inset ${this.progressCount * 400}px 0 0 -50px var(--cl-primary-60)`
-    if (this.progressCount == 1) {
-      this.showDoneButton = true
-      this.innerText = this.localization.translateKey(LocKeys.BTN_DONE)
-    }
+    this.innerText = this.getFinishButtonText(this.progressCount)
+    this.shadowStyle = this.getProgressBarStyle(this.progressCount)
+    this.progressDisplay = Math.ceil(this.progressCount) * 100
+    this.etaText = this.getEtaText(this.progressCount)
```

Comment:
did you mean to use progressDisplay param here?

Reply:
Ah my bad yes it should be `progressDisplay`. Made the changes.